src/check_datapackage/check.py

```diff
@@ -48,17 +48,40 @@ def __init__(
         self,
         issues: list[Issue],
     ) -> None:
-        """Create the DataPackageError attributes from issues."""
-        # TODO: Switch to using `explain()` once implemented
-        errors: list[str] = _map(
-            issues,
-            lambda issue: f"- Property `{issue.jsonpath}`: {issue.message}\n",
-        )
-        message: str = (
-            "There were some issues found in your `datapackage.json`:\n\n"
-            + "\n".join(errors)
-        )
-        super().__init__(message)
+        """Create the DataPackageError from issues."""
+        super().__init__(explain(issues))
+
+
+def explain(issues: list[Issue]) -> str:
+    """Explain the issue in a human-readable format.
+
+    Args:
+        issues: A list of `Issue` objects to explain.
+
+    Returns:
+        A human-readable explanation of the issue.
+    """
+    issue_explanations: list[str] = _map(
+        issues,
+        _create_explanation,
+    )
+    error_message: str = (
+        f"{len(issue_explanations)} issues were found in your `datapackage.json`:\n\n"
+        + "\n".join(issue_explanations)
+    )
+    return error_message
+
+
+def _create_explanation(issue: Issue) -> str:
+    """Create an informative explanation of what went wrong in each issue."""
+    property_name = issue.jsonpath.split(".")[-1]
+    return (  # noqa: F401
+        f"At package.{issue.jsonpath[2:]}:\n"  # indexing to remove '$.'
+        "|\n"
+        f"| {property_name}: {issue.instance!r}\n"
```

Comment:
Should we wrap `property_name` in `""` to match JSON style or match Python style? :thinking: